<?xml version="1.0"?>
<launch>

  <!--===============================================--> 
  <!--  coord frames & map -->
  <!--===============================================--> 
  <!--
  <param name="robot_description" textfile= "$(find hstar_amp)/config/model.urdf"/>
  <include file="$(find hstar_amp)/launch/tf.launch"/> 
  -->

  <!-- @TODO load map -->
  <!-- <include file="$(find ht_core)/launch/load_map.launch"/>-->

  <!--===============================================--> 
  <!--  joy & parse -->
  <!--===============================================--> 
  <node name="base_joystick"
        pkg="joy"
        type="joy_node"
        output="screen">
    <param name="deadzone" value="0.05"/>
    <param name="dev" type="string" value="/dev/input/js0"/>
  </node>

  <node name="joystick_parser"
        pkg="hstar_amp"
        type="joystick_parser.py"
        output="screen">
    <param name="drive_type" value="mecanum"/>
  </node>

  <!--===============================================--> 
  <!--  lcd                                          -->
  <!--===============================================--> 
  <node name="lcd"
        pkg="hstar_amp"
        type="serial_lcd.py"
        output="screen">
    <param name="port" value="$(env HSTAR_AMP_LCD_DEVICE)"/>
    <param name="display_type" value="ip addr"/>
    <param name="dim_timeout" value="3.0"/>
    <param name="refresh_rate" value="1.0"/>
    <param name="enable_dimming" value="false"/>
    <param name="init_display_time" value="3.0"/>
    <param name="use_rosserial" value="true"/>
  </node>

  <!-- @TODO rosserial -->

</launch>
